src/libNetwork/P2PComm.cpp

```diff
@@ -771,6 +986,261 @@ void P2PComm::AcceptConnectionCallback([[gnu::unused]] evconnlistener* listener,
+    LOG_GENERAL(DEBUG, "P2PSeed BEV_EVENT_CONNECTED");
+  }
+  if (events & BEV_EVENT_ERROR) {
+    LOG_GENERAL(WARNING, "Error: P2PSeed BEV_EVENT_ERROR");
+  }
+  if (events & BEV_EVENT_READING) {
+    LOG_GENERAL(DEBUG, "P2PSeed BEV_EVENT_READING");
+  }
+  if (events & BEV_EVENT_WRITING) {
+    LOG_GENERAL(DEBUG, "P2PSeed BEV_EVENT_WRITING ");
+  }
+  if (events & BEV_EVENT_EOF) {
+    LOG_GENERAL(DEBUG, "P2PSeed BEV_EVENT_EOF");
+  }
+  if (events & (BEV_EVENT_EOF | BEV_EVENT_ERROR)) {
+    LOG_GENERAL(DEBUG, "P2PSeed BEV_EVENT_EOF or BEV_EVENT_ERROR");
+    bufferevent_set_timeouts(bev, NULL, NULL);
+    CloseAndFreeBevP2PSeedConn(bev);
+  }
+  if (events & BEV_EVENT_TIMEOUT) {
+    LOG_GENERAL(DEBUG, "P2PSeed BEV_EVENT_TIMEOUT");
+    CloseAndFreeBevP2PSeedConn(bev);
+  }
+}
+
+void P2PComm ::ReadCbClientSeed(struct bufferevent* bev,
+                                [[gnu::unused]] void* ctx) {
+  // TODO Remove or comment logmarker later on
+  LOG_MARKER();
+  int fd = bufferevent_getfd(bev);
+  struct sockaddr_in cli_addr {};
+  socklen_t addr_size = sizeof(struct sockaddr_in);
+  getpeername(fd, (struct sockaddr*)&cli_addr, &addr_size);
+  Peer from(cli_addr.sin_addr.s_addr, cli_addr.sin_port);
+
+  // Get the data stored in buffer
+  struct evbuffer* input = bufferevent_get_input(bev);
+  if (input == NULL) {
+    LOG_GENERAL(WARNING, "Error: bufferevent_get_input failure.");
+    CloseAndFreeBevP2PSeedConn(bev);
+    return;
+  }
+  size_t len = evbuffer_get_length(input);
+  if (len == 0) {
+    LOG_GENERAL(WARNING, "Error: evbuffer_get_length failure.");
+    CloseAndFreeBevP2PSeedConn(bev);
+    return;
+  }
+  bytes message(len);
+  if (evbuffer_copyout(input, message.data(), len) !=
+      static_cast<ev_ssize_t>(len)) {
+    LOG_GENERAL(WARNING, "Error: evbuffer_copyout failure.");
+    CloseAndFreeBevP2PSeedConn(bev);
+    return;
+  }
+
+  if (message.size() <= HDR_LEN) {
+    LOG_GENERAL(WARNING, "Error: Empty message received.");
+    CloseAndFreeBevP2PSeedConn(bev);
+    return;
+  }
+
+  const uint32_t messageLength =
+      (message[4] << 24) + (message[5] << 16) + (message[6] << 8) + message[7];
+
+  {
+    // Check for length consistency
+    uint32_t res;
+
+    if (!SafeMath<uint32_t>::sub(message.size(), HDR_LEN, res)) {
+      LOG_GENERAL(WARNING, "Error: Unexpected subtraction operation!");
+      CloseAndFreeBevP2PSeedConn(bev);
+      return;
+    }
+
+    if (res > messageLength) {
+      LOG_GENERAL(WARNING,
+                  "Error: Received msg len is greater than header msg len")
+      CloseAndFreeBevP2PSeedConn(bev);
+      return;
+    } else if (res < messageLength) {
+      return;
+    }
+  }
+
+  unique_ptr<struct bufferevent, decltype(&CloseAndFreeBevP2PSeedConn)>
+      socket_closer(bev, CloseAndFreeBevP2PSeedConn);
+
+  if (evbuffer_drain(input, len) != 0) {
+    LOG_GENERAL(WARNING, "Error: evbuffer_drain failure.");
+    return;
+  }
+
+  const unsigned char version = message[0];
```

Comment:
May be move the checks for version and chain id much before

Reply:
As discussed, i think we can still move it up so that we get to know of wrong version/chainid well before rest of message is read.

Reply:
done